### ISSUE TITLE
Can't publish a Maui app that references a class library on Windows without modifying the project

### ISSUE THREAD

**Issue by javiercn** [dotnet/maui#8304]
### Description
A Maui app that references a class library can't be published without modifying the project file using the following command:

```console
msbuild /restore /t:Publish /p:TargetFramework=net6.0-windows10.0.19041 /p:TargetFrameworks="net6.0-windows10.0.19041;net6.0" /p:configuration=release /p:GenerateAppxPackageOnBuild=true /p:AppxPackageSigningEnabled=true /p:PackageCertificateThumbprint="YOUR_CERT_THUMBPRINT_HERE" /p:PackageCertificatePassword="AnyTestPasswordYouLike"
```

### Steps to reproduce
1. Create a Maui application.
2. Create a class library.
3. Reference the class library from the Maui application.
4. Open a developer command prompt.
5. Navigate to the Maui app folder.
6. Run the command above to publish the app.

### Workaround
* Update the project file with the following changes
   ```diff
   -               <TargetFrameworks>net6.0-android;net6.0-ios;net6.0-maccatalyst</TargetFrameworks>
   -               <TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('windows'))">$(TargetFrameworks);net6.0-windows10.0.19041.0</TargetFrameworks>
   +               <TargetFramework>net6.0-windows10.0.19041.0</TargetFramework>
   ```
* Update the command to use the following:
  ```console
  msbuild /restore /t:Publish /p:configuration=Release /p:GenerateAppxPackageOnBuild=true /p:AppxPackageSigningEnabled=true /p:PackageCertificateThumbprint="<<YOURTHUMBPRINTHERE>>" /p:PackageCertificatePassword="AnyTestPasswordYouLike"
  ```

**Comment by dsplaisted** [comment]
I believe if you remove the following from the command line (without modifying the project file), this should work:

`/p:TargetFrameworks="net6.0-windows10.0.19041;net6.0"`

**Comment by javiercn** [comment]
@dsplaisted unfortunately it doesn't because it also has the ios and android TFMs

**Comment by veikkoeeva** [comment]
I suppose this the same problem I have.

1. Open `Visual Studio 2022, version 17.3.0 Preview 3.0`.
2. Using VS, create a Maui project.
3. Add a .NET 6 class library the the solution using VS.
4. Reference this newly created .NET 6 class library in `MauiProgram`.

The class library project reference can be added to the Maui project. Using VS it is possible to navigate from navigate to it (`<Ctrl+F12>`). However when compiling an error is generated:
> error CS0246: The type or namespace name 'TestClassLibrary' could not be found.

Here `TestClassLibrary` being whatever one used when generating the project.

The class library `.csproj` is as follows
```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net6.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>

</Project>
```

**Comment by EmilyFeng97** [comment]
@javiercn This issue does not repro when change the class library .csproj is `<TargetFrameworks>net6.0;net6.0-windows</TargetFrameworks>`, but there is a warning `Path to mspdbcmf.exe could not be found`. 
And the https://github.com/aspnet/AspNetCore-ManualTests/pull/1519 has update the Blazor Hybrid Sample Project TFM. 
So I want to confirm with you, should I need file a new bug to track this warning and close this issue?
![image](https://user-images.githubusercontent.com/66412474/182350052-d8ef9f3f-7ac9-4030-b7f2-bb838eeb78d1.png)
"C:\Users\v-xinyfe\source\repos\MauiApp1\MauiApp1\MauiApp1.csproj" (Publish target) (1:10) ->
(_EnsurePdbCmfExeFullPath target) ->
  C:\Users\v-xinyfe\.nuget\packages\microsoft.windowsappsdk\1.1.3\buildTransitive\Microsoft.Build.Msix.Packaging.target
s(341,5): warning : Path to `mspdbcmf.exe` could not be found. A symbols package will not be generated. Review https://
aka.ms/windowsappsdkdocs and ensure that all prerequisites for Windows App SDK development have been installed. [C:\Use
rs\v-xinyfe\source\repos\MauiApp1\MauiApp1\MauiApp1.csproj]

**Comment by javiercn** [comment]
@EmilyFeng97 Just to be very clear, the issue is that you should be able to have a class library with just `<TargetFramework>net6.0</TargetFramework>`. What you are describing is at best a workaround.

I can't speak to the warning that you are seeing because that comes from the Windows SDK. If this is for manual testing purposes, I would stick to the workaround that I provided above.

**Comment by veikkoeeva** [comment]
I cross-reference https://github.com/dotnet/maui/issues/9091. I think this works for me now with the latest VS but there's similarish issue. That linked issue seem to be a show stopper if it is a supposed scenario, a bit similar to this and seem to be related around tooling with Maui and .NET 6.

**Comment by jfversluis** [comment]
@javiercn as mentioned in the [Docs](https://docs.microsoft.com/dotnet/maui/windows/deployment/overview#publish) for this

> Attempting to publish a .NET MAUI solution will result in the dotnet publish command attempting to publish each project in the solution individually, which can cause issues when you've added other project types to your solution. Therefore, the dotnet publish command should be scoped to your .NET MAUI app project.

Try specifically to specify the .NET MAUI csproj to publish. Additionally, you should be able to use `dotnet publish` now for the Windows target as well. 

Let us know if that worked!

**Comment by javiercn** [comment]
@jfversluis I was not publishing the solution. I was publishing the Maui project. I do not think that calling MSBuild vs MSBuild Core (dotnet publish) makes a difference here.

**Comment by jfversluis** [comment]
@javiercn the command you stated above does not reflect specifying the csproj. If you do not specify that explicitly it will try to publish the sln (or maybe each project in that folder individually, not really sure). You will have to specify the csproj in the command explicitly like so: `msbuild MyApp.csproj /restore /t:Publish /p:TargetFramework=net6.0-windows10.0.19041 /p:configuration=release /p:GenerateAppxPackageOnBuild=true /p:AppxPackageSigningEnabled=true /p:PackageCertificateThumbprint="YOUR_CERT_THUMBPRINT_HERE" /p:PackageCertificatePassword="AnyTestPasswordYouLike"`

**Comment by javiercn** [comment]
> 5\. Navigate to the Maui app folder.

Are you saying that it will walk up the directory hierarchy and try to build the solution?

**Comment by jfversluis** [comment]
Let's put it this way: I have never seen it work if you don't specify the csproj explicitly :)

This might be a good test case and if it appears to be that, let's find out where to log a but to maybe change that before for the dotnet CLI tooling

**Comment by jfversluis** [comment]
Hm that's strange then. In that case I need to dive into it again for a bit! Thanks for the investigation on your part!

**Comment by EmilyFeng97** [comment]
@javiercn This issue also repro when publishing a Maui app that references a class library using the following command in Dev17.4 Preview 2 installed from [PR412410](https://devdiv.visualstudio.com/DevDiv/_git/VS/pullrequest/412410). If I manually add "net6.0-android" to the SampleContent.csproj file and publish, this issue does not repro. Should I file a new bug to track this issue?
`dotnet publish -f:net6.0-android -c:Release`
![image](https://user-images.githubusercontent.com/66412474/185597494-c939e256-e906-4e3e-8e5c-b749ac629ae4.png)

**Comment by TanayParikh** [comment]
> installed from [PR412410](https://devdiv.visualstudio.com/DevDiv/_git/VS/pullrequest/412410).

Does this still reproduce using the "normal" build of VS you utilize for validations. 

> installed from [PR412410](https://devdiv.visualstudio.com/DevDiv/_git/VS/pullrequest/412410)

We're still ironing out some details for that build, so it's not the best measure of whether or not a bug is still around 😄

**Comment by EmilyFeng97** [comment]
@TanayParikh This issue repro on Dev17.4 Preview 2 installed from main branch when the MAUI project is created via CLI. And this issue does not repro if the MAUI project is created in VS. 
In windows testing scenarios, the MAUI project is usually created in VS. I created the project via CLI due to issue https://github.com/aspnet/AspNetCore-ManualTests/issues/1565. So should we ignore this issue?

**Comment by TanayParikh** [comment]
> So should we ignore this issue?

Thanks for the clarification. No we shouldn't ignore this issue if it repros on `Dev17.4 Preview 2 installed from main branch`. I just wanted to confirm it wasn't specific to the PR branch 😄

**Comment by EmilyFeng97** [comment]
> Thanks for the clarification. No we shouldn't ignore this issue if it repros on `Dev17.4 Preview 2 installed from main branch`. I just wanted to confirm it wasn't specific to the PR branch 😄

This issue is not specific to the PR branch, it also repro on VS installed from main branch when the MAUI project is created via CLI. So should I file a new issue now about publishing projects created via the CLI, or file a new issue when the issue still repro after #8304 is fixed？

**Comment by TanayParikh** [comment]
Thanks for the details @EmilyFeng97. Let's use this issue to track that as well. If after this issue is closed, projects created via CLI but published using VS are still failing, please create a new issue.

**Comment by marcelbeeker** [comment]
Any news about this issue? dotnet build -c Release -f net6.0-android failed when building a .NET MAUI containing several class libraries targeting net6.0 results in the same error.

**Comment by TanayParikh** [comment]
@marcelbeeker which version of the SDK are you on (`dotnet --version`).

I believe @jinzhao1127 may have mentioned this no longer reproduces.

**Comment by jinzhao1127** [comment]
> I believe @jinzhao1127 may have mentioned this no longer reproduces.

Yes, this issue does not repro anymore on both 6.0 and 7.0 Blazor Maui app that referenced razor class library app

**Comment by marcelbeeker** [comment]
The SDK version I use is version 6.0.402

Verzonden vanuit Mail<https://go.microsoft.com/fwlink/?LinkId=550986> voor Windows

Van: Tanay ***@***.***>
Verzonden: donderdag 13 oktober 2022 00:16
Aan: ***@***.***>
CC: ***@***.***>; ***@***.***>
Onderwerp: Re: [dotnet/maui] Can't publish a Maui app that references a class library on Windows without modifying the project (Issue #8304)


@marcelbeeker<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmarcelbeeker&data=05%7C01%7C%7C2545c7444ac2442022ef08daac9f60ed%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C638012097767312629%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=ELR9Myk2CPzTIox7mPtNdyJhcM9Xguiw%2FyYcR%2Fp37hI%3D&reserved=0> which version of the SDK are you on (dotnet --version).

I believe @jinzhao1127<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjinzhao1127&data=05%7C01%7C%7C2545c7444ac2442022ef08daac9f60ed%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C638012097767312629%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=aANsmu%2BUJQ%2FZ8%2Bc6coLbAnlEjmoHLguhC2Elg4kzsQc%3D&reserved=0> may have mentioned this no longer reproduces.

**Comment by marcelbeeker** [comment]
I create a traditional Maui (XAML) app (not Blazor).

I referenced a Class library project targeting .NET6 (net6.0).

Verzonden vanuit Mail<https://go.microsoft.com/fwlink/?LinkId=550986> voor Windows

Van: Jin ***@***.***>
Verzonden: donderdag 13 oktober 2022 08:04
Aan: ***@***.***>
CC: ***@***.***>; ***@***.***>
Onderwerp: Re: [dotnet/maui] Can't publish a Maui app that references a class library on Windows without modifying the project (Issue #8304)


I believe @jinzhao1127<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjinzhao1127&data=05%7C01%7C%7C87daa7eaf2af4fbe328508daace0c610%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C638012378635560619%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=RfSFAbrqNXu%2FsMJxgA703WBatBRKMm2xSqAjzou9t3w%3D&reserved=0> may have mentioned this no longer reproduces.

Yes, this issue does not repro anymore on both 6.0 and 7.0 Blazor Maui app that referenced razor class library app

**Comment by marcelbeeker** [comment]
Today I discovered this issue is caused by my Test projects. When I included these NuGet packages in my Test project, the errors begin.

First I have to include multi targeting in all of my projects like this:

<TargetFrameworks>net6.0;net6.0-android;net6.0-ios;net6.0-maccatalyst</TargetFrameworks><TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('windows'))">$(TargetFrameworks);net6.0-windows10.0.19041.0</TargetFrameworks>

After when I put these lines, the errors went away. But, when I included the NuGet packages below in my test project the receive these errors again.

<PackageReference Include="xunit" Version="2.4.2" />
<PackageReference Include="xunit.runner.visualstudio" Version="2.4.5">
<PrivateAssets>all</PrivateAssets>
<IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
</PackageReference>.

The errors I receive are these:

C:\Program Files\dotnet\packs\Microsoft.NET.Runtime.MonoTargets.Sdk\6.0.10\Sdk\RuntimeComponentManifest.targets(3,5): error : Empty ResolveFrameworkReference.RuntimePackPath while trying to read runtime components manifest. ResolvedFrameworkReference available: { Microsoft.NETCore.App, RuntimePackPath:  }
1>Done building project "MyFirstMaui,Tests.csproj" -- FAILED.
1>C:\Program Files\dotnet\packs\Microsoft.NET.Runtime.MonoTargets.Sdk\6.0.10\Sdk\RuntimeComponentManifest.targets(3,5): error : Empty ResolveFrameworkReference.RuntimePackPath while trying to read runtime components manifest. ResolvedFrameworkReference available: { Microsoft.NETCore.App, RuntimePackPath:  }
1>Done building project "MyFirstMaui,Tests.csproj" -- FAILED.
In all my projects in de solution I have to target to multiple 

Unfortunately, I can't find a solution for this.

**Comment by Kazbek** [comment]
Have next steps:
1) Create default .NET MAUI project
2) Create .NET 6 class library (with just one static string field to use in MAUI project)
3) Add lib reference to .NET MAUI project

Publish (via wizard):
For Debug|Any CPU is ok
For Release|Any CPU (only difference in projects):
Cannot resolve Assembly or Windows Metadata file '...\MauiAppTestPublish\ClassLibrary\bin\Release\net6.0\ClassLibrary.dll'

**Comment by j03ybh** [comment]
@Kazbek , Exactly that is my problem too. I cannot even add one .NET 6 project class library to my Maui Blazor app, and this exact same warning is presented to me. The build-solution even enters a state where it never completes.

**Comment by heron1** [comment]
I'm curious but isn't the whole point of .NET to self contain everything in the one sdk? Shouldn't it all just work without any extra fiddling around and additional dependencies?

**Comment by TranNguyen17893** [comment]
I am facing the same problem as https://github.com/dotnet/maui/issues/8768 - considered as duplicate here, when publish in iOS.
My MAUI Blazor project, with a Blazor class library:
Can run in Debug mode, for both simulator and real device.
For iOS, can build success  in Release mode, but gets errors on publish.
For Android, haven't tried yet.
Im using lasted VS2022 Version 17.4.3, Xcode 14.2 (the problem happend before I update VS2022 to the lastest version, I can't remember the old version)

build command
`dotnet publish -f:net7.0-ios -c:Release /p:ServerAddress=abc /p:ServerUser=abc /p:ServerPassword=abc /p:RuntimeIdentifier=ios-arm64 /p:TcpPort=58181 /p:ArchiveOnBuild=true /p:_DotNetRootRemoteDirectory=/Users/tranviettravel/Library/Caches/Xamarin/XMA/SDKs/dotnet/ `

error log
```
 Determining projects to restore...
  All projects are up-to-date for restore.
  Executing SayHello Task to establish a connection to a Remote Server.
                        Properties:
                                SessionId=79c3425643240xxxdeff14f87a3cf3c1443c1f155c3b556852dad0bbc62e6368,
                                Addresss=abc,
                                SshPort=22,
                                TcpPort=,
                                User=abc,
                                AppName=emTransViet,
                                VisualStudioProcessId=,
                                ContinueOnDisconnected=true
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\16.1.1481\tools\msbuild\iOS\Xamarin.Shared.targets(1738,3): error : Could not find Microsoft.iOS in /Users/tranviettravel/Library/Caches/Xamarin/XMA/SDKs/dotnet/packs/Microsoft.iOS.Sdk/16.1.1481/. [C:\EventManage\
emTransViet\emTransViet.csproj::TargetFramework=net7.0-ios]
C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\16.1.1481\tools\msbuild\iOS\Xamarin.Shared.targets(1738,3): error :          [C:\EventManage\emTransViet\emTransViet.csproj::TargetFramework=net7.0-ios]
```
on MAUI Blazor app .csproj
```
 <PropertyGroup>
    <TargetFrameworks>;net7.0-ios;net7.0-android</TargetFrameworks>
..............
    <SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'ios'">14.2</SupportedOSPlatformVersion>
    <SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'android'">24.0</SupportedOSPlatformVersion>
</PropertyGroup>
<PropertyGroup Condition="$(TargetFramework.Contains('-ios')) and '$(Configuration)' == 'Release'">
  <RuntimeIdentifier>ios-arm64</RuntimeIdentifier>
</PropertyGroup>
```
on the class library  .csproj, I tried to add **;net7.0-ios;net7.0-android**, but do not work, it has only 1 TargetFramework net7.0 as default.
```
  <PropertyGroup>
    <TargetFrameworks>net7.0;net7.0-ios;net7.0-android</TargetFrameworks>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>
```
The sdk folder above **/Users/tranviettravel/Library/Caches/Xamarin/XMA/SDKs/dotnet/packs/Microsoft.iOS.Sdk/16.1.1481/**  exists in my Mac, with contents inside.
![Screenshot 2022-12-21 185700](https://user-images.githubusercontent.com/11607187/208899865-c78f4e2e-2844-46bf-93bf-7c2e89c1c3da.png)

**Comment by techscgh** [comment]
I have created a related ticket:
[https://github.com/dotnet/maui/issues/13869](url)

**Comment by hiten247** [comment]
Faced this issue again in VS2022 17.6.5. Fix was to close VS, then delete the <project>.csproj.user file followed by the hidden ".vs"  folder at solution path. Start VS, open project and rebuild. Issue resolved.

**Comment by servetvural** [comment]
Here is a weird solution. 
Create 2 projects, 1 as Maui App and 1 as Class Library. Set both projects to have same namespace regardless the project names.
Put dbcontext class and all the other classes in class library. Run ef core functions using class library project as the startup project. This will generate migration folder and migrations. Now copy migration folder and all the classes, including the dbcontext into Maui App Project. Remove Class Library project from the solution and do as you like compiling or packaging the Maui Project. No need to mess around with the project targets etc.
I did it worked. Enjoy.

**Comment by rodrigojuarez** [comment]
I'm getting this for .Net 7 even from github actions
Any workarounds or fixes?

**Comment by Zhanglirong-Winnie** [comment]
Verified this issue with Visual Studio Enterprise 17.9.0 Preview 1.1. Not repro this issue.

**Comment by AndreasReitberger** [comment]
Still an issue on `net8 8.0.3`. Just create a maui app from the VS template and added a simple class lib reference. 
Publishing using VS => right click publish fails with the exact same error.

**Comment by FarazXTS** [comment]
I was facing the same issue when publishing my .net 8 MAUI ios app to AppStore. I fixed it by setting the target framework to net8.0-ios for all of my C# class libraries as well:
    
`    <TargetFramework>net8.0-ios</TargetFramework>`

**Comment by SaeedAtEposcompany** [comment]
I'm facing similar problem. I got a maui project X referencing a class library Y which is referencing another class library Z which is referencing Newtonsoft.Json (13.0.1) from the Microsoft.AspNet.WebApi.Client 6.0 package. 
Any CPU (Debug & Release) builds ok.
when I try to publish for windows I get "the type or namespace name 'Newtonsoft' could not be found" error.
TargetFrameworks of all project in the solution are set to net8.0

am not sure if the following is related:
I could sucessfully publish the project when I had the TargetFrameworks set to net7.0
Any help appreciated

**Comment by rodrigojuarez** [comment]
So, we can't use any CI/CD for MAUI apps targeting window that references other custom class libraries? Any workarounds?

**Comment by marcelbeeker** [comment]
Yes we can. I have a working AzureDevops pipeline which publish the MAUI app to AppCenter

Verzonden vanaf Outlook voor Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by rodrigojuarez** [comment]
@marcelbeeker can you link any examples of your configuration for the pipeline?
And regarding AppCenter, what type of application are you using? I wasn't even aware you could use appcenter with Windows MAUI apps (I don't even see the option in the new app dialog)

**Comment by marcelbeeker** [comment]
I've build a Android .MAUI application. Below you will find my build pipeline. I'll hope this is enough information for you to solve your issues.



# Variable 'KeyStorePassword' was defined in the Variables tab
trigger:
  branches:
    include:
    - refs/heads/master
    - refs/heads/development
name: $(date:yyyyMMdd)$(rev:.r)
jobs:
- job: Phase_1
  displayName: Phase 1
  cancelTimeoutInMinutes: 1
  pool:
    vmImage: windows-latest
  steps:
  - checkout: self
  - task: ***@***.***
    displayName: Use .NET Core sdk 8.x
    inputs:
      version: 8.x
  - task: ***@***.***
    inputs:
      nuGetServiceConnections: 'MyCryptoProvider'
  - task: ***@***.***
    displayName: Install Maui
    inputs:
      script: >
        dotnet workload install maui --configfile nuget.config
  - task: ***@***.***
    name: NuGetToolInstaller2
    displayName: Use NuGet 6.x
    inputs:
      versionSpec: 6.x
      checkLatest: true
  - task: ***@***.***
    name: NuGetCommand3
    displayName: NuGet restore
    inputs:
      solution: '**\*.sln'
      feedRestore: 613d6b9d-285e-48ec-a9ce-004b0a3321ff/7cc32ed4-d773-4d44-8aaf-5d1f764362a4
      searchPatternPush: $(Build.ArtifactStagingDirectory)/*.nupkg
  - task: ***@***.***
    displayName: dotnet publish
    inputs:
      command: publish
      publishWebProjects: false
      arguments: SecondMobileApp/SecondMobileApp.MAUI/SecondMobileApp.MAUI.csproj -c Release -f net8.0-android --no-restore
      zipAfterPublish: false
      modifyOutputPath: false
  - task: ***@***.***
    displayName: Copy Signed.abb file
    inputs:
      SourceFolder: $(agent.builddirectory)
      Contents: '**/*-Signed.aab'
      TargetFolder: $(build.artifactstagingdirectory)
      flattenFolders: true
  - task: ***@***.***
    displayName: dotnet test
    inputs:
      command: test
      projects: '**\*.sln'
      arguments: -f net7.0
  - task: ***@***.***
    name: PublishBuildArtifacts8
    displayName: 'Publish Artifact: drop'
    inputs:
      PathtoPublish: $(build.artifactstagingdirectory)
      TargetPath: '\\my\share\$(Build.DefinitionName)\$(Build.BuildNumber)'
...





Sent from Outlook<http://aka.ms/weboutlook>

________________________________
